### ISSUE TITLE
Migrating from Certbot to Acme.sh

### ISSUE THREAD

**Issue by danjde** [acmesh-official/acme.sh#3294]
Hi Devs!

On Debian/Apache2 VPSs, I would like to substitute "certbot" with your acme.sh, and whit me other my collaborators, due the continuous requests for updates and very strict policies on use.

Now I'm asking, as a person who does not yet know your software well, if this migration can be "painless".
Given that, I which come from certbot (or certbot-auto) have (and would like to keep) own cumulative certificates (all certs in one) into a single tld domain directory:

`"/etc/letsencrypt/live/myserver.org/"`

and on each virtualhost this specific piece of code:

```
Alias /.well-known/acme-challenge/ /var/www/letsencrypt/ #shorter
RewriteRule ^/?(.*) https://%{SERVER_NAME}/$1 [R,L]
```

Coming from certbot, could it be right to proceed in this way? ->

1. obtain the certs:

```
acme.sh  --issue  \
-d aa.com  -w /var/www/aa.com/public_html \
-d bb.com  -w /var/www/bb.com/public_html \
-d cc.com  -w /var/www/cc.com/public_html \
-d dd.com  -w /var/www/dd.com/public_html \
-d ee.com  -w /var/www/ee.com/public_html
```

2. install crontab (with acme.sh executable in /usr/bin/acme and certs in tld domain server directory "/etc/letsencrypt/live/myserver.org/" ):
`0 0 * * * "/usr/bin/acme"/acme.sh --cron --home "/etc/letsencrypt/live/myserver.org/" > /dev/null`

Could be right?
Many thanks!

Davide
Italy

**Comment by Neilpang** [comment]
you don't need to migrate.
Just remove the cronjob for certbot first.

Spend about 5 minutes reading our usage here: 
https://github.com/acmesh-official/acme.sh



>> and on each virtualhost this specific piece of code:
```
Alias /.well-known/acme-challenge/ /var/www/letsencrypt/ #shorter
RewriteRule ^/?(.*) https://%{SERVER_NAME}/$1 [R,L]
```

If you are using apache server,  you don't need to add this piece of code to your virtualhost.

Just use:
```
acme.sh  --issue -d  example.com  --apache
```

You will get a cert. 

Then copy the cert and key to where you want:

```
acme.sh --install-cert  ......
```

See: 

https://github.com/acmesh-official/acme.sh/wiki/Deploy-ssl-certs-to-apache-server

**Comment by danjde** [comment]
Hi Neilpang and thanks fro your help!
I had also opened [a post on Letsencrypt community](https://community.letsencrypt.org/t/replace-certbot-auto-with-acme-sh/139973/11), because it also seems useful to further spread your solution, which never hurts ;-)
At the same time, I had the opportunity to explore other useful aspects of your shell script.

I am left in doubt whether is it possible to install acme.sh in /usr/local/bin or similar linux binary path. What do you think about?

Thanks again.

Davide

**Comment by danjde** [comment]
Hi Neilpang,
I've installed acme.sh and obtained the first certs with it, all works fine!
However I've a doubt, that I ask you if possible to clarify me: 

now the certs are into the home user, I've installed them running manually

./acme.sh  --install-cert -d my.server.tld --cert-file /etc/certificates/my.server.tld/cert.pem --key-file /etc/certificates/my.server.tld/privkey.pem --fullchain-file /etc/certificates/my.server.tld/fullchain.pem --reloadcmd "systemctl reload apache2

but for the next renewal, should I add to crontab the "install" command?

And if so, eventually adding the "--renew-hook" option?

Thanks again!
